src/environmentd/src/lib.rs

```diff
@@ -586,6 +585,13 @@ impl Listeners {
             (adapter_storage, audit_logs_handle)
         };
 
+        // Enable Persist compaction if we're not in read only.
+        if read_only {
+            config.controller.persist_clients.cfg().disable_compaction();
+        } else {
```

Comment:
```suggestion
        if !read_only {
```

Reply:
I find it marginally clearer to rely on the fact that compaction stats out disabled!